lib/configuration/read.js

```diff
@@ -166,6 +167,28 @@ const parseConfigurationFile = async (configurationPath) => {
         );
       }
     }
+    case '.mjs': {
+      try {
+        require.resolve(configurationPath);
+      } catch {
+        throw new ServerlessError(
+          `Cannot load "${path.basename(configurationPath)}": File not found`,
+          'CONFIGURATION_NOT_FOUND'
+        );
+      }
+      try {
+        // dynamic import might not be available depending on the node version of the user, but we assume it is, since
+        // the user explicitly declared his use of es-module syntax by using the .mjs extension.
+        return await import(`file://${configurationPath}`).then((module) => module.default);
```

Comment:
I've just did small refactor, and introduced [dedicated `import-esm` utility](https://github.com/serverless/serverless/blob/e791e046e231a6d49c7cfb32cdeed0cad241fa41/lib/utils/import-esm.js) (so we don't have to relax ESLint rules for whole file). Let's rely on that

Reply:
Changed that part to use the utility function.

Just a quick note though: The prefix `file:///` should be `file://` instead. While equivalent when providing an absolute path afterwards, automatically prefixing the file root can lead to some really hard to track behavior if a non-absolute path is ever passed as an argument.

As an example, given `./some/file.js` would make `file:///./some/file.js` which would try to resolve `/some/file.js` from the users file system root.

Reply:
@RianFuro that's a very good point. I've improved that at: https://github.com/serverless/serverless/pull/11593

_So far it wasn't problematic, as in all cases we were submitting _absolute_ paths to the util, and that extra `/` wasn't harmful (I believe it came in unnoticed when first introduced)._

Reply:
I don't believe that it would ever have been problematic, but why leave that door open if it's easy to close :grin: